eventsCommon.cpp

```diff
@@ -1684,9 +1682,24 @@ EventResolveField(
 							 tmpStringBuffer, _countof(tmpStringBuffer), FALSE );
 			}
 		}
-
-		EventSetFieldS( EventBuffer, FieldIndex, _tcsdup( tmpStringBuffer ), TRUE );
+#elif defined __linux__
+        void                    *eventPtr;
+        PTCHAR                  imagePath;
+    
+        switch(EventType->EventId){
```

Comment:
The design behind sysmon events is such that each event where hashing is relevant has a m_HashType field that is set to the hashing type. We should use the same design for Linux as well to keep the code as consistent as possible. Upstream to this code (perhaps in handleEvent in Sysmon repo for process create and file delete) you could set the m_HashType field and then in this piece of code you would only have to #ifdef the GetFileHash (calling LinuxGetFileHash in the Linux case). Below is the GetHashTypeInformation function from the Windows code base (it's pretty straightforward) and you could add that to the SysmonCommon repo (#ifdef non applicable windows code). 

`//--------------------------------------------------------------------
//
// GetHashTypeInformation
//
// Get hash type information
//
//--------------------------------------------------------------------
BOOL
GetHashTypeInformation(
    __in PSYSMON_EVENT_TYPE_FMT EventType,
	__in_opt PSYSMON_EVENT_HEADER EventHeader,
	__in PULONG *HashType,
	__in PULONG FileIndex
	)
{
	if( EventHeader == NULL ) {

		return FALSE;
	}
	
	if( EventType == &SYSMONEVENT_CREATE_PROCESS_Type ) {

		*HashType = &EventHeader->m_EventBody.m_ProcessCreateEvent.m_HashType;
		*FileIndex = F_CP_Image;
	} else if( EventType == &SYSMONEVENT_FILE_DELETE_Type ) {

		*HashType = &EventHeader->m_EventBody.m_FileDeleteEvent.m_HashType;
		*FileIndex = F_FD_TargetFilename;

	} else {

		return FALSE;
	}

	//
	// If it is not defined, default to current
	//
	if( **HashType == 0 ) {

		**HashType = SysmonCryptoCurrent();		// Default is sha256 
	}

	return TRUE;
}`

Reply:
@eeriedusk  If you want, I can add the missing function since there might also be some other small pieces here and there that need to be moved. Let me know and I can add a commit to your branch.